application_store/db/queries/application/queries.py

```diff
@@ -266,25 +269,96 @@ def submit_application(application_id) -> Applications:
             "jsonb_blob": ApplicationSchema().dump(application),
             "type_of_application": application_type,
         }
-        stmt = postgres_insert(AssessmentRecord).values([row])
 
-        upsert_rows_stmt = stmt.on_conflict_do_nothing(index_elements=[AssessmentRecord.application_id]).returning(
-            AssessmentRecord.application_id
+        existing_application = db.session.scalar(
+            select(AssessmentRecord)
+            .where(
+                AssessmentRecord.application_id == row["application_id"],
+                AssessmentRecord.is_withdrawn == False,  # noqa: E712
```

Comment:
Not too sure, but can we not avoid the typing problem by using the `not_` from `sqlalchemy`?

Reply:
or maybe `is False`, if that works (I'm not sure)?

Reply:
I think it didn't recognise that it's legitimate SQLAlchemy syntax (https://stackoverflow.com/questions/18998010/flake8-complains-on-boolean-comparison-in-filter-clause) 

...anyway I bended to the will of the linter and used a workaround.